### DISCUSSION TITLE
AD LDAP(S) user federation high availability?

### DISCUSSION THREAD

**Discussion by zxcvxzcv-johndoe** [keycloak/keycloak#26259] [Q&A] [answered]
Hello,

How can I configure Keycloak to use secondary DC's for LDAP(S) user federation? 

My connection URL is the FQDN of the AD domain (i.e. contoso.local) and it returns IP-addressess of all the DC's but Keycloak seems to only use the first one it can connect to? 

For example if I shutdown the first DC it doesnt seem to automatically try to connect to other IP's of the contoso.local

**Comment by darius-m** [comment, marked as answer]
I think there may be an issue with DNS resolution in this case.
LDAP seems to fail over correctly if the connect URL looks like `ldap://server-1.localdomain:389 ldap://server-2.localdomain:389` (the two are separated by spaces).

**Reply by zxcvxzcv-johndoe** [reply]
Thanks! That seems to be working.
